modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_info_query.rb

```diff
@@ -0,0 +1,157 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages::Peripherals::StorageInteraction::Nextcloud
+  class FilesInfoQuery
+    using Storages::Peripherals::ServiceResultRefinements
+
+    FILES_INFO_PATH = 'ocs/v1.php/apps/integration_openproject/filesinfo'.freeze
+
+    def initialize(storage)
+      @uri = URI(storage.host).normalize
+      @oauth_client = storage.oauth_client
+    end
+
+    def call(user:, file_ids: [])
+      if file_ids.nil?
+        return Util.error(:error, 'File IDs can not be nil', file_ids)
+      end
+
+      if file_ids.empty?
+        return ServiceResult.success(result: [])
+      end
+
+      Util.token(user:, oauth_client: @oauth_client) do |token|
+        files_info(file_ids, token).map(&parse_json) >> handle_failure >> create_storage_file_infos
+      end
+    end
+
+    private
+
+    def files_info(file_ids, token)
+      ServiceResult.success(
+        result: RestClient::Request.execute(
+          method: :post,
+          url: Util.join_uri_path(@uri, FILES_INFO_PATH),
+          payload: { fileIds: file_ids }.to_json,
+          headers: {
+            'Authorization' => "Bearer #{token.access_token}",
+            'Accept' => 'application/json',
+            'Content-Type' => 'application/json',
+            'OCS-APIRequest' => true
+          }
+        )
+      ).map(&:body)
+    rescue RestClient::Unauthorized => e
+      Util.error(:not_authorized, 'Outbound request not authorized!', e.response)
+    rescue RestClient::NotFound => e
+      Util.error(:not_found, 'Outbound request destination not found!', e.response)
+    rescue RestClient::ExceptionWithResponse => e
+      Util.error(:error, 'Outbound request failed!', e.response)
+    rescue StandardError
+      Util.error(:error, 'Outbound request failed!')
+    end
+
+    def parse_json
+      ->(response_body) do
+        # rubocop:disable Style/OpenStructUse
+        JSON.parse(response_body, object_class: OpenStruct)
+        # rubocop:enable Style/OpenStructUse
+      end
+    end
+
+    def handle_failure
+      ->(response_object) do
+        if response_object.ocs.meta.status == 'ok'
+          ServiceResult.success(result: response_object)
+        else
+          Util.error(:error, 'Outbound request failed!', response_object)
+        end
+      end
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    def create_storage_file_infos
+      ->(response_object) do
+        ServiceResult.success(
+          result: response_object.ocs.data.to_h.map do |key, value|
+            if value.statuscode == 200
+              ::Storages::StorageFileInfo.new(
```

Comment:
Yes, but as it's a class and needs to have that many attributes because it's basically a DTO. It needs to be like that.

Addition: I will try to resolve this issue with `Data.define`.